### PR TITLE
SGA-7804 lambda for monitoring snowflake datastore

### PR DESCRIPTION
**Pull request by guyscher2**

### CODE REVIEW

File: monitoring-datastores/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+- Blackbox monitoring of your Satori Datastore.
+
+## How does it work?
+- Python script which runs a simple query through Satori and throws exception if fails.
+- Python code deployed as an AWS Lambda function.
+- Amazon EventBridge triggers the Lambda every hour.
+
+## Alerting
+- The python code by default raise an exeption when query fails.
+  You may capture the expetion with CloudWatch and Incident Manager or replace the exeption in the code with one of our alert snippets (alerts.py).
```
Comment (alexku7) [inline monitoring-datastores/README.md]: expetion :)

File: monitoring-datastores/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+## Networking
+- If you use Satori to whitelist IP's, you will need to attach the lambda to a NAT gateway and whitelist his IP.
```
Comment (alexku7) [inline monitoring-datastores/README.md]: whitelist **its** IP.

File: monitoring-datastores/snowflake/main.tf [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  build_in_docker  = true
+
+  timeout = 60
+
+  environment_variables = {
+    snowflake_user = "",
+    snowflake_password = "",
+    snowflake_account = "",
+    snowflake_satori_host = "",
+    snowflake_warehouse = ""
```
Comment (alexku7) [inline monitoring-datastores/snowflake/main.tf]: You don't want to pass it as variables and explain to a customer how to pass it in secured way ?

File: monitoring-datastores/README.md (hunk 1 above)
Comment (guyscher2) [reply monitoring-datastores/README.md]: exception
image

File: monitoring-datastores/alerts.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
+from slack_sdk import WebClient
+import os, requests
+
+def sendSlackAlert():
+    SLACK_BOT_TOKEN = os.environ['SLACK_BOT_TOKEN']
+    slack_client = WebClient(SLACK_BOT_TOKEN)
+    channel_name = os.environ["channel_name"]
```
Comment (tomershaniii) [inline monitoring-datastores/alerts.py]: why are some env vars lower and some upper?

File: monitoring-datastores/alerts.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+def createOpsgenieIncident():
+    response = requests.post(
+        "https://api.opsgenie.com/v1/incidents/create",
+        headers={
+        "Content-Type": "application/json",
+        "Authorization": f"GenieKey {os.environ['opsgenie_token']}"
+        },
+        data={
+            "message": "Datastore not responding",
+            "description":"Blackbox monitor for Datastore has failed!",
```
Comment (tomershaniii) [inline monitoring-datastores/alerts.py]: Isn't it the the alerting that's failed?

File: monitoring-datastores/postgresql/README.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
+## How does it work?
```
Comment (tomershaniii) [inline monitoring-datastores/postgresql/README.md]: same as the top level, let's keep only datastore specific stuff here

File: monitoring-datastores/postgresql/main.tf [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+module "monitor_postgresql_datastore_lambda_function" {
+  source = "terraform-aws-modules/lambda/aws"
+  
+  function_name = "monitor-postgresql-datastore-lambda"
+  description   = "Lambda function that monitors postgresql datastore"
+  handler       = "monitor-postgresql-datastore.main"
+  runtime       = "python3.10"
+
+  publish                  = true
+  architectures = ["arm64"]
```
Comment (tomershaniii) [inline monitoring-datastores/postgresql/main.tf]: do we have to specify arch?

File: monitoring-datastores/snowflake/README.md [hunk 8]
```diff
@@ -0,0 +1,18 @@
+## How does it work?
+- Python script which runs a simple query through Satori and throws exception if fails.
+- Python code deployed as an AWS Lambda function.
```
Comment (tomershaniii) [inline monitoring-datastores/snowflake/README.md]: common...

File: monitoring-datastores/postgresql/main.tf (hunk 7 above)
Comment (guyscher2) [reply monitoring-datastores/postgresql/main.tf]: We build the code on docker then push it for lambda.
In the docker build process we downloads a pip lib called psycopg2.
For some reason it downloads it for arm64 so I must run the lambda on it as well.

Maybe for customer the docker build will be different. if so he can just delete that line.

**Review by alexku7 — APPROVED** [review]

**Pull request merged**